pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -82,43 +81,38 @@ func (esc *endpointSliceController) updateEDS(es interface{}, event model.Event)
 					// For service without selector, maybe there are no related pods
 				}
 
-				var labels map[string]string
-				locality := getLocalityFromTopology(e.Topology)
-				sa, uid := "", ""
+				var initEndpoint model.IstioEndpoint
 				if pod != nil {
-					sa = kube.SecureNamingSAN(pod)
-					uid = createUID(pod.Name, pod.Namespace)
-					labels = configKube.ConvertLabels(pod.ObjectMeta)
+					podCopy := *pod
```

Comment:
Another sign the abstraction we created doesn't match our use case. I don't think we want to be making copies like this

Reply:
I only want to mutate the pod labels, donot need deepcopy. This is really tricky here

Reply:
fixed